Observability updates

dockerymick opened the pull request:
Making lots of updates to the install topic. Also working on adding the tasks to customize the install 

- [Enable/disable](https://github.com/open-cluster-management/backlog/issues/4621)
- [Manage observability](https://github.com/open-cluster-management/backlog/issues/4775)
- [Uninstall](https://github.com/open-cluster-management/backlog/issues/5236)

dockerymick committed:
Adding info to manage observability

dockerymick committed:
updating install topic, adding uninstall

dockerymick committed:
added example of observability file

dockerymick committed:
adding more to the observability topic

dockerymick committed:
adding more

dockerymick committed:
more updates on configuring

dockerymick committed:
Update observe_intro.adoc

Made some updates

dockerymick committed:
updating RBAC by adding info for observability

dockerymick committed:
small change

dockerymick committed:
Update observe_intro.adoc

dockerymick committed:
Update observe_intro.adoc

dockerymick committed:
Update manage_observe.adoc

dockerymick committed:
Update manage_observe.adoc

making progress

dockerymick committed:
Update manage_observe.adoc

more changes, still waiting on review

dockerymick committed:
Update manage_observe.adoc

Adding example of alert manager and reference link for parameter explanations

dockerymick committed:
Update observability_install.adoc

dockerymick committed:
Update observability_install.adoc

clyang82 commented on observability/manage_observe.adoc:
```diff
@@ -1,9 +1,116 @@
+
+. Log in to your {product-title-short} hub cluster.
+. By default, the alert rules are defined in the `thanos-rule-default-rules` configuration map in the `open-cluster-management-observability` namespace. You can create multple rules in the configuration.
+
```
> By default, the alert rules are defined in the `thanos-rule-default-rules` configuration map in the `open-cluster-management-observability` namespace. You can create multple rules in the configuration

Suggest to:
By default, the OOTB alert rules are defined in the `thanos-ruler-default-rules` ConfigMap in the `open-cluster-management-observability` namespace. You can create the `thanos-ruler-custom-rules` ConfigMap in the `open-cluster-management-observability` namespace to define multiple rules.

dockerymick replied:
What does OOTB mean?

robdolares replied:
@dockerymick - 'Out of the Box'

dockerymick replied:
Thanks!

clyang82 commented on observability/manage_observe.adoc:
```diff
@@ -1,9 +1,116 @@
+
+. Log in to your {product-title-short} hub cluster. 
+. From the navigation menu, select *Observe environments* > *Overview* > *Grafana link*
+. You can access the Grafana dashboard from the _Clusters_ page. From the navigation menu, select *Automate infrastructure* > *Clusters* > *Grafana link*.
```
Suggest to mixin the above 2 items. for example:
From the navigation menu, select *Observe environments* > *Overview* > *Grafana link* (You can also access the Grafana dashboard from the _Clusters_ page. From the navigation menu, select *Automate infrastructure* > *Clusters* > *Grafana link*)

BTW, How about just use `Grafana` instead of `Grafana link`?

robdolares replied:
Agree with @clyang82 on combining, or at least an 'You can also' from new line if you have to.
Also, should we pluralize _dashboard_ --> 'dashboards'?

> "You can access the Grafana dashboards from..."

You do land on the overview dashboard, but there are numerous to choose from or drill down to.

dockerymick replied:
I wasn't sure if there were multiple dashboards. Thanks for clarifying. I update dashboard to dashboards

clyang82 commented on observability/manage_observe.adoc:
```diff
@@ -1,9 +1,116 @@
+    enableMetrics: false #indicates the observability addon push metrics to hub server
+----
+
+When you disable observability, the `multicluster-observability-operator` continues to run on the managed cluster, while the `metrics-collector` deployment is scaledd to zero. When a managed cluster with the observability component is detached, the `metrics-collector` deployments are removed.
```
s/scaledd/scaled

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -1,29 +1,15 @@
-. Create and configure object storage for the observability service. Your object storage might resemble the following information:
-
+- You must have {product-title} hub cluster installed as a managed cluster to use the observability feature.  
+- You must install Grafana version 6.0 or later. For more information, see https://grafana.com/docs/grafana/latest/installation/[Install Grafana].
```
no need to install Grafana in Prerequisites

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -1,29 +1,15 @@
-
-. Create and configure object storage for the observability service. Your object storage might resemble the following information:
-
+- You must have {product-title} hub cluster installed as a managed cluster to use the observability feature.  
```
That is not required. Suggest just mention ACM installed required.
The order I think is:
1. OCP installed
2. ACM installed.
3. Create Object storage.

dockerymick replied:
I made the statement about ACM according to this comment: https://github.com/open-cluster-management/backlog/issues/4966#issuecomment-687171918

Should the statement for 4966 be added as a known issue instead? @subbarao-meduri may you help clarify ?

subbarao-meduri replied:
@dockerymick - It is not a requirement that a hub cluster must be installed as a managed cluster. 
We can document as a known issue that if a hub cluster is not configured as managed cluster, ACM will not collect metrics from the hub cluster. @bjoydeep please chime in with any additional comments.

bjoydeep replied:
Agreed with @subbarao-meduri and @clyang82 . It is NOT needed for Observability to function.

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -34,9 +20,29 @@ config:
+[#enablement-for-observability]
+== Enablement for observability
+
+Enable the observability service by creating a `multicluster-observability` CustomResourceDefinition (CRD). Complete the following steps to enable the observability service: 
```
s/`multicluster-observability` CustomResourceDefinition (CRD)/MultiClusterObservability CustomResourceDefinition instance

dockerymick replied:
updated `multicluster-observability` to MultiClusterObservability CustomResource (CR) instance

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -34,9 +20,29 @@ config:
+kubectl get pods -n open-cluster-management|grep observability
+----
+.. For the `observability-addons` run the following command:
++
```
remove the check for `observability-addons`

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -57,22 +63,85 @@ oc get secret multiclusterhub-operator-pull-secret -n open-cluster-management --
 ----
```
> Generate a pull-secret for the `observability` namespace by running the following command:

Generate a pull-secret for the `open-cluster-management-observability` namespace by running the following command

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -57,22 +63,85 @@ oc get secret multiclusterhub-operator-pull-secret -n open-cluster-management --
+----
 
-. Create a MultiClusterObservability resource by running the following command:
+. Create a `MultiClusterObservability` CRD by running the following command:
```
s/`MultiClusterObservability` CRD/`MultiClusterObservability` instance or resource

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -57,22 +63,85 @@ oc get secret multiclusterhub-operator-pull-secret -n open-cluster-management --
 ----
++
+Your `MultiClusterObservability` CRD might resemble the following file:
++
```
Not CRD, it is CR. so just use instance or resource here

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -57,22 +63,85 @@ oc get secret multiclusterhub-operator-pull-secret -n open-cluster-management --
 
 +
 ----
 kubectl create -f mco_cr.yaml
```
remove `kubectl create -f mco_cr.yaml` . create/apply do the same thing.

robdolares committed:
typo  'Prometheus'

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -57,22 +63,85 @@ oc get secret multiclusterhub-operator-pull-secret -n open-cluster-management --
+----
 
-//still working to add steps
+After you install observability, all of the pods in your `open-cluster-management-observability` namespace for your object storage, Grafana, and AlertManager are automatically installed. All managed clusters that are connected to {product-title-short} can send metrics to the cluster.
```
s/your object storage/Thanos

dockerymick replied:
I think object storage is appropriate bc it is more generic. To my understanding Thanos is not a required storage solution. 

I can update the following statement: 

"After you install observability, all of the pods in your `open-cluster-management-observability` namespace for your object storage, Grafana, and AlertManager are automatically installed."

TO 

"Based on the previous resource example, all of the pods in your `open-cluster-management-observability` namespace Thanos, Grafana, and AlertManager are automatically installed after you install observability."

Thoughts?

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -57,22 +63,85 @@ oc get secret multiclusterhub-operator-pull-secret -n open-cluster-management --
+Uninstall the observability service by completing the following steps:
+
+. Log in to your hub cluster.
+. Delete the `multicluster-observability-operator` resource instance with the following command:
```
s/multicluster-observability-operator/MultiClusterObservability

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -57,22 +63,85 @@ oc get secret multiclusterhub-operator-pull-secret -n open-cluster-management --
+. Delete the `multicluster-observability-operator` resource instance with the following command:
++
+----
+kubectl delete multicluster-observability-operator observability
```
s/multicluster-observability-operator/MultiClusterObservability

clyang82 commented on observability/observe_intro.adoc:
```diff
@@ -1,24 +1,22 @@
+Observability is configured with Prometheus AlertManager to communicate with third-party alerting systems. You can also create custom alerting rules. Explore your managed cluster data, such as the control plane health, optimization data, and cluster resources from Grafana.
 
-The observability service requires data storage for the time series data. You must configure your object store to enable the observability service. {product-title-short} supports different types of storage solutions. 
+The observability service requires data storage for the time series data. You must configure your object store to enable the observability service. {product-title-short} supports different types of storage solutions, such as link:https://docs.aws.amazon.com/whitepapers/latest/aws-overview/storage-services.html[Amazon S3], link:https://cloud.google.com/storage/docs/introduction[Google Cloud Storag ...
```
We did not have tested the `Google Cloud Storage` and `Azure storage`. Suggest to just mention the `Amazon S3` only here. for example:
such as link:https://docs.aws.amazon.com/whitepapers/latest/aws-overview/storage-services.html[Amazon S3]. for more options, refer to link:https://thanos.io/v0.2/thanos/storage.md/[Thanos].

dockerymick replied:
Thanks for clarifying. I added the other storage options due to the statement from the What's new presentation. I will slack the link to the presentation

clyang82 replied:
Let us add other storage options since we have them in What's new presentation. Thanks @dockerymick 
But we should remove Thanos which is not a storage.

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -57,22 +63,85 @@ oc get secret multiclusterhub-operator-pull-secret -n open-cluster-management --
+kubectl delete multicluster-observability-operator observability
+----
+
+When you delete the resource, the `multicluster-observability-operator` that run on {product-title-short} hub cluster, and the `multicluster-observability-addon` that run on all managed clusters are removed. 
```
> When you delete the resource, the `multicluster-observability-operator` that run on {product-title-short} hub cluster, and the `multicluster-observability-addon` that run on all managed clusters are removed. 

Suggestion:
When you delete the resource, the pods are in `open-cluster-management-observability` namespace on {product-title-short} hub cluster, and the pods in `open-cluster-management-addon-observability` namespace on all managed clusters are removed.

clyang82 commented:
Thanks @dockerymick I leave some comments. One general question: Do we have a tool to do grammar check for doc?

dockerymick commented:
@clyang, thanks for your review. Previously, we used Acrolinx but no longer have access to the licenses. I believe we can find an open source tool plugin

clyang82 commented:
@dockerymick I am not if we evaluate https://github.com/client9/misspell or not.

dockerymick committed:
Update manage_observe.adoc

updates after the review

dockerymick committed:
Update observability_install.adoc

more updates after the review

dockerymick committed:
Update observe_intro.adoc

Only commented the links out for now , Waiting on clarification based on the What's new 2.1 presentation

dockerymick committed:
Update manage_observe.adoc

more updates

dockerymick committed:
Update manage_observe.adoc

updated configuration map to ConfigMap

dockerymick committed:
Update manage_observe.adoc

dockerymick committed:
Update observability_install.adoc

Add info to view observability installation status

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -57,22 +55,125 @@ oc get secret multiclusterhub-operator-pull-secret -n open-cluster-management --
+. View the installation status of the observability service for each managed cluster by running the following command:
++
+----
+oc get managedclusteraddon -n test observability-controller -o yaml
```
Where is `test` namespace from? suggest to use `local-cluster` or just mentioned that `test` is a managed cluster namespace. the name should be `observability-addon` instead of `observability-controller`. @marcolan018 correct me if I am wrong

dockerymick committed:
Update observability_install.adoc

adding a link reference to the troubleshoot topic for observability

dockerymick committed:
Update observability_install.adoc

Update test to local-cluster

dockerymick committed:
Update observe_intro.adoc

Adding other cloud storage options

dockerymick committed:
Update observability_install.adoc

Removed Thanos and added Amazon S3

dockerymick committed:
Update manage_observe.adoc

dockerymick committed:
Update manage_observe.adoc

still making edits

dockerymick committed:
Update manage_observe.adoc

more updates

dockerymick committed:
Update observability_install.adoc

continuing to review

dockerymick committed:
Update observability_install.adoc

adding comment

dockerymick committed:
Update observability_install.adoc

come back to verify that no comments are left and links work

dockerymick committed:
Update rbac.adoc

come back to revise observability rbac

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -116,7 +116,7 @@ Based on the previous resource example, all of the pods in your `open-cluster-ma
 +
 ----
-oc get managedclusteraddon -n test observability-controller -o yaml
+oc get managedclusteraddon -n local-cluster observability-controller -o yaml
```
change it to `oc get observabilityaddon -n local-cluster observability-addon -o yaml`

marcolan018 replied:
@clyang82 managedclusteraddon is the correct name. it's unify addon status report resource provided by klusterlet

clyang82 commented on observability/observability_install.adoc:
```diff
@@ -128,7 +128,7 @@ metadata:
   creationTimestamp: "2020-10-19T19:19:32Z"
   generation: 1
   name: observability-controller
```
s/observability-controller/observability-addon

clyang82 replied:
```
apiVersion: observability.open-cluster-management.io/v1beta1
kind: ObservabilityAddon
```

marcolan018 replied:
same as above. Don't need to change to observability-addon @clyang82

dockerymick committed:
Update observe_intro.adoc

Move: When the observability service is installed, the following CustomResourceDefinitions (CRDs) are included `multicluster-observability-operator`, `multicluster-observability-addon`, `observatoria.core.observatorium.io`.

to the enablement section 

We only support the  data storage listed in Thanos table cloud

dockerymick committed:
Update observability_install.adoc

Come back to this to review

dockerymick committed:
Update observability_install.adoc

updates after review w/Joydeep

dockerymick committed:
Update rbac.adoc

updates after review with joydeep

dockerymick committed:
Update observe_intro.adoc

updates

swopebe committed:
Wow! Few small changes you can find in my commit. I didn't doc test but that is bc I know you do when you can. Congrats on nearly-perfect new content!

swopebe committed:
few small changes, another good one.

swopebe commented on observability/observe_intro.adoc:
```diff
@@ -6,21 +6,17 @@ You can use {product-title} to visualize and monitor metrics from your clusters
 == Observability service
 
-Observability is installed on the hub cluster during the product installation. The observability service can be installed in standard or high availability configurations. The observability service uses the `metrics-collector` pod on your managed cluster to collect data with OpenShift Prometheus, and sends it to {product-title-short}. 
+Observability is installed, but not enabled during the product installation. The observability service uses the `metrics-collector` pod on your managed cluster to "scrape" data from OpenShift Prometheus, and sends it to {product-title-short} hub cluster. 
```
I would say what we say in components about "default".

By default, obs is Installed but not enabled...? Or something like that because we emphasize default behavior in other areas of the doc.

swopebe commented on security/rbac.adoc:
```diff
@@ -271,4 +272,19 @@ View the following console and API RBAC tables for Governance lifecycle:
+[#observability-api-RBAC]
+=== Observability RBAC
+
+To use the observability features: access managed cluster metrics, use the search component, use the Visual Web Terminal (VWT), a user must have access to the managed cluster. To create,update, and delete the MultiClusterObservability custom resource, the user must have the `cluster-admin` role or the `open-cluster-management:cluster-manager-admin` role. View the following RBAC table:
```
This one gets me a bit. Seems you are listing actions. but it gets buried... and the words may not be parallel. Maybe:

- Access managed cluster metrics
- Use the search component (OR Search for resources -- more active but up to you)
- Use the Visual Web Terminal (NO VWT--we are not using that abbreviation) if you have access to the managed cluster

Then start with a new line: To create, update (needs a space here). Also, "the user" here can just be:

To complete these actions, you must be assigned ___ or __ role...    

Use punctuation on all or none.

swopebe commented on security/rbac.adoc:
```diff
@@ -271,4 +272,19 @@ View the following console and API RBAC tables for Governance lifecycle:
+| -
+|===
+
 Continue to learn more about securing your cluster, see xref:../security/security_intro.adoc#security[Security].
```
To continue..., see....

swopebe reviewed (approved):
I will approve, as I am done committing the small changes I had. Great job and congrats on seeing a new component area from end-to-end in the doc!!! 

Please see: https://github.com/open-cluster-management/rhacm-docs/pull/1383/files#r513567837 where I got a bit confused and made some suggestions.

dockerymick committed:
Update observe_intro.adoc

update after Brandi's review

dockerymick committed:
Update rbac.adoc

updates after Brandi's review

dockerymick committed:
Update rbac.adoc

updates

dockerymick commented:
All comments from reviews have been addressed, thank you everyone! I will continue to merge this PR and continue to make updates
